LayoutTests/fast/screen-orientation/orientation-types-and-angles.html

```diff
@@ -0,0 +1,111 @@
+            );
+            jsTestIsAsync = true;
+
+            const rotations = new Map([
+                [
+                    "landscape-right",
+                    {
+                        type: "landscape-secondary",
+                        angle: 270,
+                        legacy: -90,
+                    },
+                ],
+                [
+                    "portrait",
+                    {
+                        type: "portrait-primary",
+                        angle: 0,
+                        legacy: 0,
+                    },
+                ],
+                [
+                    "landscape-left",
+                    {
+                        type: "landscape-primary",
+                        angle: 90,
+                        legacy: 90,
+                    },
+                ],
+                // Would only work with .lock() enabled
+                // [
+                //     "portrait-upsidedown",
+                //     {
+                //         type: "portrait-secondary",
+                //         angle: 180,
+                //         legacy: 180,
+                //     },
+                // ],
+            ]);
+
+            function runRotationUIScript(rotation) {
+                debug(`Simulating rotation to ${rotation}.`);
+                const script = `
+                    (async () => {
+                        await new Promise(r => uiController.simulateRotationLikeSafari("${rotation}", r));
+                        await new Promise(r => uiController.doAfterVisibleContentRectUpdate(r));
+                        uiController.uiScriptComplete();
+                    })();
+                `;
+                const orientationChangeEventPromise = new Promise((r) =>
+                    screen.orientation.addEventListener("change", r, {
+                        once: true,
+                    })
+                );
+                const uiScriptPromise = new Promise((r) =>
+                    testRunner.runUIScript(script, r)
+                );
+                return Promise.all([
+                    orientationChangeEventPromise,
+                    uiScriptPromise,
+                ]);
+            }
+
+            onload = async () => {
+                try {
+                    debug("tyring to lock...")
+                    await screen.orientation.lock("any");
+                } catch (err) {
+                    debug(err.message);
+                }
+                for (const rotation of rotations.keys()) {
+                    debug(`====== ${rotation} START ======`);
+
+                    await runRotationUIScript(rotation);
+
+                    const { type, angle, legacy } = rotations.get(rotation);
+                    debug(
+                        `Expected angle: ${angle} | actual: ${screen.orientation.angle}`
+                    );
+                    angleIsCorrect = angle === screen.orientation.angle;
```

Comment:
seems we should be using:
```
shouldBe("screen.orientation.angle", "angle");
```
or
```
shouldBe("screen.orientation.angle", "" + angle);
```
if the first former one doesn't print the angle value in the output.